streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -719,10 +719,10 @@ void runOnce() {
 
         final long pollLatency = pollPhase();
 
-        // Shutdown hook could potentially be triggered and transit the thread state to PENDING_SHUTDOWN during #pollRequests().
-        // The task manager internal states could be uninitialized if the state transition happens during #onPartitionsAssigned().
```

Comment:
This comment and the short-circuit `return` was a fix for an NPE from a year or two ago, but it turns out we actually broke this fix when we encapsulated everything into the `pollPhase` -- [the fix](https://issues.apache.org/jira/browse/KAFKA-8620) was to return in between returning from `poll()` and calling `addRecordsToTasks()`, since we could end up with uninitialized tasks/TaskManager state if the shutdown hook was triggered during the rebalance callback. 
Luckily, at some point we happened to shore up the task management logic so that the rebalance callbacks will always proceed even if the thread has already been told to shut down, so we're not in any trouble here. This also means that technically, we don't even need to `return` here anymore -- but there's no real reason to continue through the loop, so I just updated the comment and left it as is.